memory-stores/memory-store-mongodb/deployment/src/main/java/io/quarkiverse/langchain4j/memorystore/mongodb/deployment/MongoDBMemoryStoreProcessor.java

```diff
@@ -0,0 +1,67 @@
+package io.quarkiverse.langchain4j.memorystore.mongodb.deployment;
+
+import java.util.Objects;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.ClassType;
+import org.jboss.jandex.DotName;
+
+import com.mongodb.client.MongoClient;
+
+import dev.langchain4j.store.memory.chat.ChatMemoryStore;
+import io.quarkiverse.langchain4j.memorystore.MongoDBChatMemoryStore;
+import io.quarkiverse.langchain4j.memorystore.mongodb.runtime.MongoDBMemoryStoreRecorder;
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.mongodb.MongoClientName;
+import io.quarkus.mongodb.deployment.MongoClientNameBuildItem;
+
+class MongoDBMemoryStoreProcessor {
+
+    public static final DotName MONGODB_CHAT_MEMORY_STORE = DotName.createSimple(MongoDBChatMemoryStore.class);
+    private static final String FEATURE = "langchain4j-memory-store-mongodb";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    public MongoClientNameBuildItem requestMongoClient(MongoDBMemoryStoreBuildTimeConfig config) {
+        return new MongoClientNameBuildItem(config.clientName().orElse("default"));
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public void createMemoryStoreBean(
+            BuildProducer<AdditionalBeanBuildItem> additionalBeanProducer,
+            BuildProducer<SyntheticBeanBuildItem> beanProducer,
+            MongoDBMemoryStoreRecorder recorder,
+            MongoDBMemoryStoreBuildTimeConfig buildTimeConfig) {
+        String clientName = buildTimeConfig.clientName().orElse(null);
+        AnnotationInstance mongoClientQualifier;
+        mongoClientQualifier = AnnotationInstance.builder(MongoClientName.class)
+                .add("value", Objects.requireNonNullElse(clientName, "default"))
+                .build();
```

Comment:
This is not correct. See https://github.com/quarkiverse/quarkus-langchain4j/blob/7c3d9ddbc2bcbcd0c4cfd327f6383e6e482aaa9c/memory-stores/memory-store-redis/deployment/src/main/java/io/quarkiverse/langchain4j/memorystore/redis/deployment/RedisMemoryStoreProcessor.java#L49 for how to do it properly